### ISSUE TITLE
Response to preflight request doesn't pass access control check: It does not have HTTP ok status.

### ISSUE THREAD

**Issue by SergiSC** [tajpouria/cors#15]
Hello,

I'm having problems passing the Cors access control using the **AttainCors** middleware and my backend server can't respond the petitions. I'd followed the examples and it does not work. I don't know if it's an issue or just a beginner error.

Here is a code snipped from my source code. 
```typescript
const app = new App();
const corsOptions: CorsOptions = {
  methods: 'GET,POST,DELETE,PUT',
  optionsSuccessStatus: 200,
  origin: "http://localhost:8080",
}
app.use(attainCors(corsOptions));
```

Any help is really appreciated. Thank you.

**Comment by tajpouria** [comment]
@SergiSC Can you show me router that supposed to handling this request, client request and the exact error message as well?

**Comment by SergiSC** [comment]
Hi, thanks for the response. Here you go!

### Router request:
```typescript
async function userLogin(req: Request, res: Response) {
    const options: BodyOptions = {
        contentTypes: {
            text: ["application/javascript"],
        },
    }
    if (!req.hasBody) {
        res.status(400).send({ message: `Bad request. Request has no body!` });
        return;
    }

    const reqValue: UserDTO = JSON.parse(await (await req.body(options)).value);
    const users: UserDTO[] = await User.all();
    ...
    //private code
    ...
    }

    res.status(422).send({ message: 'Invalid username or password.' });
};
```

### Client request:
```typescript
async function logIn(body: User): Promise<User | Error> {
  let result: User = new User();
  await axios.post(baseUrl + "user/login", body).then(response => {
    if (response.status !== 200) {
      throw new Error("Unauthorized");
    } else {
      result = response.data;
    }
  });
  return result;
}
```
### Error:
Access to XMLHttpRequest at 'http://localhost:3000/api/user/login' from origin 'http://localhost:8080' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.

POST http://localhost:3000/api/user/login net::ERR_FAILED

Uncaught (in promise) Error: Network Error
    at createError (createError.js?2d83:16)
    at XMLHttpRequest.handleError (xhr.js?b50d:83)

**Comment by tajpouria** [comment]
@SergiSC @aaronwlee Sorry, I can't really say what's the problem of handler with code snippet you provide, Maybe opening same issue on [attain](https://github.com/aaronwlee/Attain) help you figure it out.

**Comment by aaronwlee** [comment]
Hi :) Thanks for tagging me.

As far as I know, setting the origin option means only accepting the request from the same origin of the option. 
And it seems like you're calling the server from the crossed origin.

@tajpouria am I right?

**Comment by SergiSC** [comment]
@tajpouria Ok, I will do it.

Thanks anyway!

**Comment by aaronwlee** [comment]
@SergiSC 

I presume the server's been hosted at 8080 and the client-server is running at 3000 port. then it means you need to accept both origins using an array.

```ts
 origin: ["http://localhost:8080", "http://localhost:3000"]
```

**Comment by SergiSC** [comment]
@aaronwlee 

I'd already tried that and it didn't work :(

**Comment by tajpouria** [comment]
@aaronwlee Yeah that's right but the client is running on 8080 and cors options are fine.
@SergiSC I suggest following these steps:
1. Setup cors with no options:
```ts
const app = new App();
app.use(attainCors()); 
``` 
2. Setup a bare minimum router **after cors setup**, make a request to make sure that's nothing wrong with your app overall setup:
```ts
app.get("/", (req, res) => {
  res.status(200).send('Hi there!');
})
```
3. If that's worked, Try rewriting your login handler piece by piece and test step, If you did not handle each step properly that probably ends up to  CORS error #8

**Comment by aaronwlee** [comment]
@tajpouria I found the reason.
Attain is unlike oak, even if it's the option method requested, it's not going to respond back to the server. This means the only way to respond is manually to use the 
```ts
response.send("")
```
method.

@SergiSC It's my bad, I'll patch it ASAP

**Comment by tajpouria** [comment]
@SergiSC Thank you for reporting this issue, Problem fixed on #16!
You may also need to remove your cached module and in order to get the latest CORS module:
e.g in linux machine: `rm -rf ~/.cache/deno`

**Comment by SergiSC** [comment]
Thank you for everything @tajpouria.
Awesome job!
